File: android/src/main/java/com/getcapacitor/community/classes/results/CurrentUserIDResult.java
package com.getcapacitor.community.classes.results;

import com.getcapacitor.JSObject;
import com.getcapacitor.community.interfaces.Result;
import java.util.UUID;

public class CurrentUserIDResult implements Result {

    private final UUID userID;

    public CurrentUserIDResult(UUID userID) {
        this.userID = userID;
    }

    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        if (userID != null) result.put("userID", userID);

        return result;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Results/CurrentUserIDResult.swift
import Foundation
import Capacitor

@objc public class CurrentUserIDResult: NSObject, Result {
    let currentUserID: UUID?

    init(_ currentUserID: UUID?) {
        self.currentUserID = currentUserID
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()

        if let currentUserID = currentUserID {
            result["userID"] = currentUserID.uuidString.lowercased()
        }

        return result as AnyObject
    }
}
